From a78de7c03246d33489a35de9cd5b6264d69d2557 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Wed, 18 Oct 2017 20:44:30 +0800
Subject: [PATCH 2658/5242] MLK-16606-3 arm64: dts: imx8qm-mek: add pca6416 IO
 expander support

commit  fe1956110fe83c98d7bc0e90023376d30f209c62 from
https://source.codeaurora.org/external/imx/linux-imx.git

NXP pca6416 is compatible with TI tca6416 and it's on M41 I2C bus.

Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts |   23 ++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
index bc339fe..203353a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
@@ -87,6 +87,13 @@
 
 &iomuxc {
 	imx8qm-mek {
+		pinctrl_cm41_i2c0: cm41i2c0grp {
+			fsl,pins = <
+				SC_P_M41_I2C0_SCL_M41_I2C0_SCL          0x0600004c
+				SC_P_M41_I2C0_SDA_M41_I2C0_SDA          0x0600004c
+			>;
+		};
+
 		pinctrl_fec1: fec1grp {
 			fsl,pins = <
 				SC_P_ENET0_MDC_CONN_ENET0_MDC			0x06000048
@@ -406,6 +413,22 @@
 	};
 };
 
+&i2c0_cm41 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_cm41_i2c0>;
+	status = "okay";
+
+	pca6416: gpio@20 {
+		compatible = "ti,tca6416";
+		reg = <0x20>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+};
+
 &fec2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_fec2>;
-- 
1.7.9.5

